SCJ-232: Refactor MaxSelectionSize into a constant

duncan-oxd opened the pull request:
This turns the hardcoded `5` into a constant called `ScTrialDateLimits.MaxSelectionSize` that can be easily changed to affect the front and back end.

I also updated the code that sets `bookingInfo.SelectedFairUseTrialDates` to limit the list, so people can't manually inject 6 or more dates when they post the form.

Take a look and let me know if the variable/file naming is okay! I originally had it as "ScTrialDateSelection" but that's already used for other things throughout the app 😆

molund commented on database/Constants/ScTrialDateLimits.cs:
```diff
@@ -0,0 +1,8 @@
+namespace SCJ.Booking.Data.Constants
+{
+    public static class ScTrialDateLimits
```
Looks great, but can we generalize the name of this file so it's an obvious place to put homeless constants?

For example, on WorkBC I had a file called `General.cs`
https://github.com/bcgov/workbc-jb/blob/master/src/WorkBC.Shared/Constants/General.cs

Maybe the variable could be called something like `ScMaxTrialDateSelections`?

duncan-oxd replied:
I'm renaming the file to `ScConstants` -- I thought about `ScGeneralConstants` but if it's already in the `Constants` namespace, hopefully it'll be clear enough. I'm not sure how that manifests in the editor!

molund reviewed (approved):
LGTM

duncan-oxd commented:
I renamed the file/variables to be more general-purpose. We're likely to have more miscellaneous constants in the future, but probably no more that are related to the "selection size limit" 👍 

I moved some of the hardcoded HTML into template strings so they'd be easier to pass down from the _Trial.cshtml file

duncan-oxd commented:
There are indeed some npm packages for converting an int to a word, but as expected they're needlessly large files, so I didn't use one for this one number.
